Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -58,7 +58,7 @@ public extension Array {
     ///   - keyPath: keyPath indicating the property that the array should be sorted by
     /// - Returns: sorted array.
     func sorted<T: Hashable>(like otherArray: [T], keyPath: KeyPath<Element, T>) -> [Element] {
-        let dict = otherArray.enumerated().reduce(into: [:]) { $0[$1.element] = $1.offset }
+        let dict = Dictionary(uniqueKeysWithValues: otherArray.enumerated().map { ($1, $0) })
```

Comment:
Could you use `lazy` here? Would it speed things up even more?
```suggestion
        let dict = Dictionary(uniqueKeysWithValues: otherArray.enumerated().lazy.map { ($1, $0) })
```

Reply:
Tested it myself, it makes virtually no difference.